dspace-api/src/test/java/org/dspace/content/BundleTest.java

```diff
@@ -739,6 +739,22 @@ public void testSetOrder() throws SQLException, AuthorizeException, FileNotFound
         assertThat("testSetOrder: Bitstream 3 is now first", bitstreams.get(0).getName(), equalTo(bs3.getName()));
         assertThat("testSetOrder: Bitstream 1 is now second", bitstreams.get(1).getName(), equalTo(bs.getName()));
         assertThat("testSetOrder: Bitstream 2 is now third", bitstreams.get(2).getName(), equalTo(bs2.getName()));
+
+        // Now give only a partial list of bitstreams
+        newBitstreamOrder = new UUID[] { bsID1, bsID2 };
+        bundleService.setOrder(context, b, newBitstreamOrder);
+
+        // Assert Bitstream order is unchanged
+        List<Bitstream> bitstreamsAfterPartialData = b.getBitstreams();
+        assertThat("testSetOrder: Partial data doesn't change order", bitstreamsAfterPartialData, equalTo(bitstreams));
```

Comment:
In line 737 you get the `bitstreams` object by calling `b.getBitstreams()`. `b` is a bundle. The `getBitstream()` method from the Bundle class gives you a *reference* to its bitstreams list. I don't know what happens if the `BundleService` changes the order of the bitstreams, but I wouldn't be surprised if it changes the original list as well. As you hold a reference on this list only, you are probably comparing two references of the same object. I think it would help to call `g.getBitstreams().toArray()`.